specs/lightclient/p2p-interface.md

```diff
@@ -0,0 +1,101 @@
+
+## Table of contents
+
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Modifications in HF1](#modifications-in-hf1)
+  - [The gossip domain: gossipsub](#the-gossip-domain-gossipsub)
+    - [Topics and messages](#topics-and-messages)
+      - [Global topics](#global-topics)
+        - [`sync_committee_contribution_and_proof`](#sync_committee_contribution_and_proof)
+      - [Sync committee subnets](#sync-committee-subnets)
+        - [`sync_committee_{subnet_id}`](#sync_committee_subnet_id)
+      - [Sync committees and aggregation](#sync-committees-and-aggregation)
+  - [The discovery domain: discv5](#the-discovery-domain-discv5)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+# Modifications in HF1
+
+## The gossip domain: gossipsub
+
+Gossip meshes are added in HF1 to support the consensus activities of the sync committees. Validators use an aggregation scheme to balance the processing and networking load across all of the relevant actors.
+
+### Topics and messages
+
+Topics follow the same specification as in the phase 0 document. The only change is the addition of new gossipsub topics added in HF1.
+
+The specification around the creation, validation, and dissemination of messages has not changed from the Phase 0 document.
+
+The new topics along with the type of the `data` field of a gossipsub message are given in this table:
+
+| Name | Message Type |
+| - | - |
+| `sync_committee_contribution_and_proof` | `SignedContributionAndProof` |
+| `sync_committee_{subnet_id}` | `SyncCommitteeSignature` |
+
+Definitions of these new types can be found in the [HF1 validator guide](./validator.md#containers).
+
+#### Global topics
+
+HF1 adds one global topic to propagate partially aggregated sync committee signatures to all potential proposers of beacon blocks.
+
+##### `sync_committee_contribution_and_proof`
+ 
+This topic is used to propagate partially aggregated sync committee signatures to be included in future blocks.
+
+The following validations MUST pass before forwarding the `signed_contribution_and_proof` on the network; define `contribution_and_proof = signed_contribution_and_proof.message` and `contribution = contribution_and_proof.contribution` for convenience:
+
+- _[IGNORE]_ The contribution's slot is for the current slot, i.e. `contribution.slot == current_slot`.
+- _[REJECT]_ The aggregate signature is valid for the message `beacon_block_root` and aggregate pubkey derived from the participation info in `aggregate_bits`.
+- _[IGNORE]_ The block being signed over (`contribution.beacon_block_root`) has been seen (via both gossip and non-gossip sources).
+- _[IGNORE]_ There has been no other valid sync committee contribution for the declared `slot` by the validator referenced by `contribution_and_proof.aggregator_index`.
```

Comment:
i think we want to remove this condition though right?

aiui aggregates other than the first will still consume gossipsub resources via the relatively expensive `IHAVE/IWANT` fetching, rather than just forwarding

Reply:
I was meaning that the `(slot, aggregator_index)` is a better measure to filter than the hash-tree-root of contribution, if multiple selected aggregators turn out to make the same contribution.

Duplicates (same contribution, different aggregator) are expensive but necessary, since they have different message IDs due to the gossip aggregator info, and we should not score those down. Changing the gossip message ID function would be better, but let's address that later.

Duplicates with same aggregator however are plain wrong, and should be ignored like in current spec. Maybe even rejected (since they shouldn't exist, there should be one per aggregator per slot right?).